### PR TITLE
docs: add example with update

### PR DESCRIPTION
**Pull request by Borda**
Adding example for update example it was not trivial to get it done, see my question #1872

### CODE REVIEW

**Comment by Borda** [comment]
@sileht @jd mind have look? ^^

File: docs/source/examples.rst [hunk 1]
```diff
@@ -529,5 +529,28 @@ request.
...
+          - -draft # filter-out GH draft PRs
+          - label="Ready-to-Go"
+        actions:
+          update: {}
+
+.. note::
+
+    Mergify is event driven bot so even you add this cool rule, it wound be executed till each PR is updated
+     or the bot is pinged in a comment as `@Mergifyio refresh`.
+
```
Comment (jd) [inline docs/source/examples.rst:554]: We should write that somewhere else in a more "general" place.
```suggestion

```

File: docs/source/examples.rst [hunk 2]
```diff
@@ -529,5 +529,28 @@ request.
               {{ check_conditions }}
 
 
+🐍 Updating PR for linear history
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As GitHub recently introduced linear history in PR setting it become very handy to use an automatic tool
+ to keep your PRs up-to-date. On the other hand you do not want to overkill your CI with building all PRs
+ always especially when there is still work in progress, so you can limit this action just to ready labeled PRs.
```
Comment (jd) [inline docs/source/examples.rst:537]:
```suggestion
As GitHub supports linear history in pull request settings, it is very handy to use a rule
to keep your pull requests up-to-date. As you do not want to trigger your CI too often by always re-running it on every pull request — especially when there is still work in progress — you can limit this action to labeled pull requests.
```

**Review by jd — CHANGES_REQUESTED** [review]
Thanks a lot!
Suggested some rewording, LGTM otherwise.
If you can just amend your commit that'd be good.

File: docs/source/examples.rst (hunk 1 above)
Comment (Borda) [reply docs/source/examples.rst:554]: so at the beginning or where bc till I knew it caused me lost of troubles to understands what is happening

**Comment by Borda** [comment]
@jd added your suggestion and moved the general note to the starter page

File: docs/source/examples.rst (hunk 1 above)
Comment (jd) [reply docs/source/examples.rst]: I'll probably add this to https://docs.mergify.io/getting-started.html

File: docs/source/getting-started.rst [hunk 3]
```diff
@@ -118,6 +118,11 @@ Here's the relevant configuration file content:
   ``merge`` action with a ``method`` parameter containing the merge method to
   use.
 
+.. note::
+
+    Mergify is event driven bot so even you add this cool rule, it wound be executed till each PR is updated
+     or the bot is pinged in a comment as `@Mergifyio refresh`.
```
Comment (jd) [inline docs/source/getting-started.rst:124]:
```suggestion
    Mergify is event-driven: rules are only executed when pull requests are updated
    or when the bot is manually triggered in a comment with `@Mergifyio refresh` (see :ref:`Commands`).
```

**Review by jd — CHANGES_REQUESTED** [review]
sounds great, reworded

**Comment by jd** [comment]
You'll need to squash your commit in a single one :)

**Comment by Borda** [comment]
> You'll need to squash your commit in a single one :)

ok, I ll do if but why you do not make squash merge here in GH UI? :P

**Comment by jd** [comment]
Haha, we usually use rebase method around here 😄 
You need to update your commit message to match the PR title (which is correct for semantic).

What we need is a `@mergifyio squash` command!

**Comment by Borda** [comment]
> What we need is a `@mergifyio squash` command!

that would be great as I think that usage is better and also easier than pushing/explaining to contributors how to make squash in cmd 🐰

**Review by jd — APPROVED** [review]

**Pull request merged**